### PR TITLE
Upgrade gradle and various build plugins

### PR DESCRIPTION
**Pull request by usmansaleem**
- Upgrade Gradle to 7.x. Allows to compile against JDK 16
- Upgrade various plugins
- Use new dependency license generation plugin with custom html renderer.

### CODE REVIEW

File: buildSrc/src/main/java/tech/pegasys/internal/license/GroupedLicensesHtmlRenderer.java [hunk 1]
```diff
@@ -0,0 +1,449 @@
...
+    }
+    output = new File(config.outputDir, fileName);
+
+    final Map<ModuleData, Set<Pair<String, String>>> moduleLicenses = buildModuleLicenses(data);
+    final Map<String, List<ModuleData>> licenseGroup = buildLicenseGroup(moduleLicenses);
+
+    writeReport(licenseGroup, moduleLicenses);
+  }
+
+  private Map<ModuleData, Set<Pair<String, String>>> buildModuleLicenses(final ProjectData data) {
```
Comment (jframe) [inline buildSrc/src/main/java/tech/pegasys/internal/license/GroupedLicensesHtmlRenderer.java:119]: Can you break this up into some smaller methods? There is a lot happening in this method

File: buildSrc/src/main/java/tech/pegasys/internal/license/GroupedLicensesHtmlRenderer.java [hunk 2]
```diff
@@ -0,0 +1,449 @@
...
+    Map<ModuleData, Set<Pair<String, String>>> moduleLicences = new TreeMap<>();
```
Comment (jframe) [inline buildSrc/src/main/java/tech/pegasys/internal/license/GroupedLicensesHtmlRenderer.java:120]: nit: final and there few others I saw too

File: buildSrc/src/main/java/tech/pegasys/internal/license/GroupedLicensesHtmlRenderer.java [hunk 3]
```diff
@@ -0,0 +1,449 @@
...
+      throw new UncheckedIOException(e);
+    }
+    return multipleLicenseMap;
+  }
+
+  private void writeReport(
+      final Map<String, List<ModuleData>> licenseGroup,
+      final Map<ModuleData, Set<Pair<String, String>>> moduleLicenses) {
+    final StringBuilder html = new StringBuilder();
+    html.append("<!doctype html>\n");
```
Comment (jframe) [inline buildSrc/src/main/java/tech/pegasys/internal/license/GroupedLicensesHtmlRenderer.java:311]: Don't particularly like html being directly in this class. Would be too much overhead to use a simple renderer like moustache?

File: buildSrc/src/main/java/tech/pegasys/internal/license/GroupedLicensesHtmlRenderer.java (hunk 1 above)
Comment (usmansaleem) [reply buildSrc/src/main/java/tech/pegasys/internal/license/GroupedLicensesHtmlRenderer.java]: moved into its own repo.

File: buildSrc/src/main/java/tech/pegasys/internal/license/GroupedLicensesHtmlRenderer.java (hunk 2 above)
Comment (usmansaleem) [reply buildSrc/src/main/java/tech/pegasys/internal/license/GroupedLicensesHtmlRenderer.java]: moved into its own repo.

File: buildSrc/src/main/java/tech/pegasys/internal/license/GroupedLicensesHtmlRenderer.java (hunk 3 above)
Comment (usmansaleem) [reply buildSrc/src/main/java/tech/pegasys/internal/license/GroupedLicensesHtmlRenderer.java]: moved into its own repo.

File: build.gradle [hunk 4]
```diff
@@ -12,20 +12,31 @@
...
+import tech.pegasys.internal.license.reporter.GroupedLicenseHtmlRenderer
 
 import java.text.SimpleDateFormat
 
+buildscript {
+  repositories {
+    maven { url "https://artifacts.consensys.net/public/maven/maven/" }
+  }
+  dependencies {
+    classpath 'tech.pegasys.internal.license.reporter:license-reporter:develop'
```
Comment (jframe) [inline build.gradle:27]: Is there a release version for the license reporter we can use?

Comment (usmansaleem) [reply build.gradle]: good catch, 1.0.0!

**Review by jframe — APPROVED** [review]
LGTM

**Pull request merged**